src/jshint.js

```diff
@@ -2260,7 +2260,13 @@ var JSHINT = (function () {
     if (!state.option.esnext) {
       warning("W119", this, "spread/rest operator");
     }
-    if (!state.tokens.next.identifier) {
+
+    // TODO: Allow all AssignmentExpression
+    // once parsing permits.
+    if (!state.tokens.next.identifier &&
+        state.tokens.next.type !== "(string)" &&
+          !checkPunctuators(state.tokens.next, ["[", "("])) {
```

Comment:
[`checkPunctuators` ensures that the `type` is `"punctuator"`](https://github.com/jshint/jshint/blob/08c70e632f889feb8481acdc41e541ca49edfad4/src/jshint.js#L4476), so the explicit type check is now unnecessary. I don't *think* that it should be necessary to check if the token is an identifier, either since if it's a punctuator it can't be an identifier.